Add React Native V6 Changes (#11320)
getsentry/sentry-docs: docs/platforms/react-native/tracing/instrumentation/expo-router.mdx

The code snippet below shows how to initialize the instrumentation.

```javascript {6-8, 15-16, 20-25} {filename: app/_layout.tsx}
import React from 'react';
import { Slot, useNavigationContainerRef } from 'expo-router';
import Constants, { ExecutionEnvironment.StoreClient } from 'expo-constants';
import * as Sentry from '@sentry/react-native';

const navigationIntegration = Sentry.reactNavigationintegration({
  enableTimeToInitialDisplay: Constants.executionEnvironment === ExecutionEnvironment.StoreClient, // Only in native builds, not in Expo Go.
});

Sentry.init({
  dsn: '___PUBLIC_DSN___',
  // Set tracesSampleRate to 1.0 to capture 100% of transactions for tracing.
  // We recommend adjusting this value in production.
  tracesSampleRate: 1.0,
  integrations: [navigationIntegration],
  enableNativeFramesTracking: Constants.executionEnvironment === ExecutionEnvironment.StoreClient, // Only in native builds, not in Expo Go.
});

function RootLayout() {
  const ref = useNavigationContainerRef();
  React.useEffect(() => {
    if (ref) {

>       navigation.registerNavigationContainer(ref);

Reviewer: I think you are referring to `navigationIntegration` instead of `navigation` here
```suggestion
      navigationIntegration.registerNavigationContainer(ref);
```

You can configure the instrumentation by passing an options object to the constructor:

```javascript
Sentry.reactNavigationintegration({
  enableTimeToInitialDisplay: true, // default: false
  routeChangeTimeoutMs: 1000, // default: 1000
  ignoreEmptyBackNavigationTransactions: true, // default: true
});
```

### routeChangeTimeoutMs

This option specifies how long the instrumentation will wait for the route to mount after a change has been initiated before the transaction is discarded. The default value is `1_000`.

### enableTimeToInitialDisplay

This option will enable automatic measuring of the time to initial display for each route. To learn more see [Time to Initial Display](/platforms/react-native/tracing/instrumentation/time-to-display). The default value is `false`.

### ignoreEmptyBackNavigationTransactions

> Does not sample transactions that are from routes that have been seen any more and don't have any spans. This removes a lot of the clutter as most back navigation transactions are now ignored. The default value is `true`.

Reviewer 2: ```suggestion
This ensures that transactions that are from routes that are no longer being seen and don't have any spans, are not being sampled. This removes a lot of clutter, making it so that most back navigation transactions are now ignored. The default value is `true`.
```
Please let me know if I changed the meaning too much here.

Author: ```suggestion
This ensures that transactions that are from routes that've been seen and don't have any spans, are not being sampled. This removes a lot of clutter, making it so that most back navigation transactions are now ignored. The default value is `true`.
```

It looks good, but I noticed I had a mistake in the original.
